assessment.sql

```diff
@@ -5,24 +5,44 @@ CREATE DATABASE "sql-crud-assessment";
 
 -- Question 1. Create a table called `licensees` to hold data in columns surname,given_name,gender,height,weight,dob,eye_color,hair_color.  See `data/licensees.csv` for example data.  Don't forget that all tables should have an id column as a unique identifier for a row.
 
--- your code here
+CREATE TABLE licensees (
+   id SERIAL PRIMARY KEY,
+   surname CHARACTER VARYING,
+   given_name CHARACTER VARYING,
+   gender CHARACTER VARYING,
+   height INTEGER,
+   weight INTEGER,
+   dob DATE,
+   eye_color CHARACTER VARYING,
+   hair_color CHARACTER VARYING
+);
 
 -- Question 2. Add `Bishop, Caren, f, 63, 132, 1943-09-26, Brown, Black` into the table of licensees.
 
--- your code here
+INSERT INTO licensees(surname,given_name,gender,height,weight,dob,eye_color,hair_color)
+  VALUES ( 'Bishop', 'Caren', 'f', 63, 132, '1943-09-26', 'Brown', 'Black' )
 
 -- Question 3. Load the data from `data/licensees.csv` into the `licensees` table.
 
--- your code here
+\copy licensees(surname,given_name,gender,height,weight,dob,eye_color,hair_color) FROM 'data/licensees.csv' WITH (HEADER true, FORMAT csv)
 
 -- Question 4. Write a query to get all attributes of licensees with Hazel eye color and Brown or Black hair color.
 
--- your code here
+SELECT COUNT(*), eye_color, hair_color
+FROM licensees
+WHERE eye_color AND hair_color = 'Brown' or 'Black';
```

Comment:
The logical operators `AND` and `OR` operate on booleans. That means that the expressions to either side of them must be something you can simplify to either `TRUE` or `FALSE`.

So, let's look at what's to the left and right of `AND`. On the left, we have `eye_color`, a column name. Can we boil that down to a boolean? No, we can't. On the right, we have `hair_color = 'Brown'`. (`AND` takes precedence over `OR`, so it happens first; we'll get to the `OR` later, if we get that far.) Can that be expressed as a boolean? Yes! It's `TRUE` if `hair_color` is 'Brown' for the current row being examined, and `FALSE` if not. 

In order for this to work, you need to rewrite this condition such that the logical operators are being fed the sort of data they want to work on. Here is a rewrite of your WHERE clause; I'd like you to step through it like in the paragraph above and understand why it's written that way.

```sql
...
WHERE eye_color = 'Hazel' AND (hair_color = 'Black' OR hair_color = 'Brown')
...
```

2/5